cmd/subnetcmd/deploy.go

```diff
@@ -228,44 +228,32 @@ func getControlKeys(network models.Network) ([]string, bool, error) {
 
 // controlKeysLoop asks as many controlkeys the user requires, until Done or Cancel is selected
 func controlKeysLoop(controlKeysPrompt string, network models.Network) ([]string, bool, error) {
-	const (
-		addCtrlKey = "Add control key"
-		doneMsg    = "Done"
-		cancelMsg  = "Cancel"
+	label := "Control key"
+	info := "Control keys are P-Chain addresses which have admin rights on the subnet.\n" +
+		"Only private keys which control such addresses are allowed to make changes on the subnet"
+	arg := network
+	list, canceled, err := app.Prompt.CaptureListDecision(
+		app.Prompt, // we need this to be able to mock test
+		controlKeysPrompt,
+		app.Prompt.CapturePChainAddress,
+		"Enter P-Chain address (Example: P-...)",
+		label,
+		info,
+		arg,
 	)
 
-	var controlKeys []string
-
-	for {
-		listDecision, err := app.Prompt.CaptureList(
-			controlKeysPrompt, []string{addCtrlKey, doneMsg, cancelMsg},
-		)
-		if err != nil {
-			return nil, false, err
-		}
-
-		switch listDecision {
-		case addCtrlKey:
-			controlKey, err := app.Prompt.CapturePChainAddress(
-				"Enter P-Chain address (Ex: `P-...`)",
-				network,
-			)
-			if err != nil {
-				return nil, false, err
-			}
-			if contains(controlKeys, controlKey) {
-				fmt.Println("Address already in list")
-				continue
-			}
-			controlKeys = append(controlKeys, controlKey)
-		case doneMsg:
-			return controlKeys, false, nil
-		case cancelMsg:
-			return nil, true, nil
-		default:
-			return nil, false, errors.New("unexpected option")
+	ctrlKeys := make([]string, len(list))
+	var (
+		key string
+		ok  bool
+	)
+	for i, k := range list {
```

Comment:
probably with go generics we can avoid this loop

Reply:
Already commented.